### PR TITLE
Upgrade `metal_irconverter` headers and bindings to `2.0.4`

### PR DESCRIPTION
**Pull request by MarijnS95**
Depends on #17

This contains all the new bindings added to the public API in [the 2.0 release](https://developer.apple.com/metal/shader-converter/#changelog).  Fortunately, as we can now see in the diff thanks to #17, there's no binary diff in the `repr(C)` structures nor used constants.

### CODE REVIEW

File: bindings_generator/vendor/metal_irconverter/metal_irconverter.h [hunk 1]
```diff
@@ -546,7 +547,7 @@ void IRCompilerSetHitgroupType(IRCompiler* compiler, IRHitGroupType hitGroupType
 
 #define IRIntrinsicMaskClosestHitAll         0x7FFFFFFF
 #define IRIntrinsicMaskMissShaderAll         0x7FFF
-#define IRIntrinsicMaskCallableShaderAll     0x7FFF
+#define IRIntrinsicMaskCallableShaderAll     0x703F
 #define IRIntrinsicMaskAnyHitShaderAll       ((uint64_t)-1)
 #define IRRayTracingUnlimitedRecursionDepth  (-1)
```
Comment (MarijnS95) [inline bindings_generator/vendor/metal_irconverter/metal_irconverter.h:552]: @FlannyH :eyes:

**Review by tosti007 — APPROVED** [review]

**Review by tosti007 — APPROVED** [review]

File: bindings_generator/src/main.rs [hunk 2]
```diff
@@ -15,7 +15,7 @@ fn compiler_bindings() {
         .header(header.to_str().unwrap())
         .parse_callbacks(Box::new(RenameCallback))
         .clang_args(&["-I", include_dir.to_str().unwrap()])
-        .dynamic_link_require_all(true)
+        .dynamic_link_require_all(false)
```
Comment (tosti007) [inline bindings_generator/src/main.rs:18]: Out of curiosity; what does this do?

**Review by tosti007 — APPROVED** [review]

File: bindings_generator/src/main.rs (hunk 2 above)
Comment (tosti007) [reply bindings_generator/src/main.rs:18]: Why does this need to be disabled?

**Comment by Jasper-Bekkers** [comment]
I've pushed the update to beta 4 to this branch, I think we should go ahead and merge it for now so we can signal that this is how it's going to be. 

We'll need to see what to do about the HACK commit, I think it's fine to check this in as a .patch file instead and apply it before generating bindings.

**Comment by MarijnS95** [comment]
> We'll need to see what to do about the HACK commit, I think it's fine to check this in as a .patch file instead and apply it before generating bindings.

We already merged this "hack commit" before as part of https://github.com/Traverse-Research/saxaboom/pull/17. It was just easier to cleanly copy the headers into the repository (undoing the change) and then cherry-picking it again (in a separate commit) and fix the conflict. Especially when then going from beta 1 to 2, 3, and 4 etc.

**Comment by MarijnS95** [comment]
Happy to merge this when CI is green. Just need to test if Windows still segfaults on Beta 4 because it started doing that on Beta 3 :(

**Pull request merged**